test/mainnet-fork/Uniswapv4.t.sol

```diff
@@ -0,0 +1,1971 @@
+        _assertZeroAllowances(address(usdt));
+    }
+
+    function _getDecreasePositionMinAmounts(
+        uint256 tokenId,
+        uint128 liquidity,
+        uint256 maxSlippage
+    ) internal returns (uint256 amount0Min, uint256 amount1Min) {
+        (
+            , // PoolKey
+            PositionInfo positionInfo
+        ) = IPositionManagerLike(_POSITION_MANAGER).getPoolAndPositionInfo(tokenId);
+
+        ( uint256 amount0Forecasted, uint256 amount1Forecasted ) = _quoteLiquidity(
+            positionInfo.tickLower(),
+            positionInfo.tickUpper(),
+            liquidity
+        );
+
+        amount0Min = (amount0Forecasted * maxSlippage) / 1e18;
+        amount1Min = (amount1Forecasted * maxSlippage) / 1e18;
+    }
+
+    function _decreasePosition(
+        uint256 tokenId,
+        uint128 liquidityDecrease,
+        uint256 amount0Min,
+        uint256 amount1Min
+    ) internal returns (DecreasePositionResult memory result) {
+        (
+            , // PoolKey
+            PositionInfo positionInfo
+        ) = IPositionManagerLike(_POSITION_MANAGER).getPoolAndPositionInfo(tokenId);
+
+        uint256 usdcBeforeCall = usdc.balanceOf(address(almProxy));
+        uint256 usdtBeforeCall = usdt.balanceOf(address(almProxy));
+
+        uint256 rateLimitBeforeCall = rateLimits.getCurrentRateLimit(_WITHDRAW_LIMIT_KEY);
+
+        uint256 positionLiquidityBeforeCall = IPositionManagerLike(_POSITION_MANAGER).getPositionLiquidity(tokenId);
+
+        vm.prank(relayer);
+        mainnetController.decreaseLiquidityUniswapV4({
+            poolId            : _POOL_ID,
+            tokenId           : tokenId,
+            liquidityDecrease : liquidityDecrease,
+            amount0Min        : amount0Min,
+            amount1Min        : amount1Min
+        });
+
+        uint256 usdcAfterCall = usdc.balanceOf(address(almProxy));
+        uint256 usdtAfterCall = usdt.balanceOf(address(almProxy));
+
+        uint256 rateLimitAfterCall = rateLimits.getCurrentRateLimit(_WITHDRAW_LIMIT_KEY);
+
+        uint256 positionLiquidityAfterCall = IPositionManagerLike(_POSITION_MANAGER).getPositionLiquidity(tokenId);
+
+        result.tokenId           = tokenId;
+        result.amount0Received   = usdcAfterCall - usdcBeforeCall;
+        result.amount1Received   = usdtAfterCall - usdtBeforeCall;
+        result.liquidityDecrease = liquidityDecrease;
+        result.tickLower         = positionInfo.tickLower();
+        result.tickUpper         = positionInfo.tickUpper();
+
+        assertGe(result.amount0Received, amount0Min);
+        assertGe(result.amount1Received, amount1Min);
+
+        assertEq(rateLimitBeforeCall - rateLimitAfterCall, _to18From6Decimals(result.amount0Received + result.amount1Received));
+
+        assertEq(IPositionManagerLike(_POSITION_MANAGER).ownerOf(result.tokenId), address(almProxy));
+
+        assertEq(positionLiquidityAfterCall, positionLiquidityBeforeCall - result.liquidityDecrease);
```

Comment:
Add assertions here as well for _getAmountsForLiquidity and assert that the amounts withdrawn correspond to the amount of liquidity that was burned